libs/barista-components/table/src/table.ts

```diff
@@ -442,6 +437,38 @@ export class DtTable<T> extends _DtTableBase<T> implements OnDestroy {
     }
     return csvObj;
 
+    /** @internal data can end up in a table many ways, figure out the best to use for exporting table data */
+    function determineDataToExport(table: DtTable<T>): T[] | null {
```

Comment:
This method needs rethinking since it is internal in another method, which uses `DS` from the body of its parent method. It isn't very easy to follow and can be simplified. 

The assumption that `DS` is always a `Datasource` is incorrect.`

`table._filteredData` is actually empty if DS was a data array and is only updated once on `connect` called under `DataSoruce`

Reply:
Refactored this a bit to hopefully be more clear and to not transparently use anything from the parent function. `DataSource` is now handled safely. I think all of the various ways data gets into a table are now handled, but please correct me if I'm missing something.